docs/content/guide/migration.ngdoc

```diff
@@ -30,6 +31,1116 @@ which drives many of these changes.
+    - [$httpBackend](guide/migration#migrate1.5to1.6-ngMock-$httpBackend)
+  - [ngResource](guide/migration#migrate1.5to1.6-ngResource)
+    - [$resource](guide/migration#migrate1.5to1.6-ngResource-$resource)
+  - [ngRoute](guide/migration#migrate1.5to1.6-ngRoute)
+    - [$route](guide/migration#migrate1.5to1.6-ngRoute-$route)
+
+
+<br />
+<a name="migrate1.5to1.6-ng-directives"></a>
+### Core: _Directives_
+
+<a name="migrate1.5to1.6-ng-directives-form"></a>
+#### **form**:
+
+<minor />
+Due to [9e24e7](https://github.com/angular/angular.js/commit/9e24e774a558143b3478536911a3a4c1714564ba),
+`FormController` now defines its methods on its prototype, instead of on each instance. As a
+consequence, `FormController` methods always need to be called in the correct context. For example
+`$scope.$watch('something', myFormCtrl.$setDirty)` will no longer work, because the `$setDirty`
+method is passed without any context. The code must now be changed to:
+
+```js
+$scope.$watch('something', function() {
+  myFormCtrl.$setDirty();
+})
+```
+
+or you can use `Function.prototype.bind` or `angular.bind`.
+
+
+<a name="migrate1.5to1.6-ng-directives-input[radio]"></a>
+#### **input[type=radio]**:
+
+<major />
+<a name="commit-5ac7da"></a>
+Due to [5ac7da](https://github.com/angular/angular.js/commit/5ac7daea72ec31cf337d1d21b13f0d17ff33994f),
+the "checked" status of radio inputs is now determined by doing a strict comparison (`===`) between
+the value of the input and the `ngModelController.$viewValue`. Previously, this was a non-strict
+comparison (`==`).
+
+This means in the following examples the radio is no longer checked:
+
+```html
+<!-- this.selected = 0 -->
+<input type="radio" ng-model="$ctrl.selected" value="0" />
+
+<!-- this.selected = 0; this.value = false; -->
+<input type="radio" ng-model="$ctrl.selected" ng-value="$ctrl.value" />
+```
+
+If your code relied on the non-strict comparison, you need to convert the values so that they
+continue to match with strict comparison.
+
+
+<a name="migrate1.5to1.6-ng-directives-input[range]"></a>
+#### **input[type=range]**:
+
+<major />
+<a name="commit-913016"></a>
+Due to [913016](https://github.com/angular/angular.js/commit/9130166767c4792c5d32d08a918fc7becf32c9a6)
+and the built-in support for range inputs, the behavior of such elements when bound to `ngModel`
+will be different than before:
+
+- Like `input[type=number]`, it requires the model to be a Number, and will set the model to a
+  Number.
+- It supports setting the min/max values only via the min/max attributes.
+- It follows the browser behavior of never allowing an invalid value. That means, when the browser
+  converts an invalid value (empty: `null`, `undefined`, `false` ..., out of bounds: greater than
+  max, less than min) to a valid value, the input will in turn set the model to this new valid value
+  via `$setViewValue`.
+  - This means a range input will never be required and never have a non-Number model value, once
```

Comment:
"This means a range input will never be required"? Should this be "will never have the required validation error"?